javascript/integration-tests/RegisterDealOrder.spec.ts

```diff
@@ -106,8 +85,70 @@ describe('RegisterDealOrder', (): void => {
                     }
                 })
                 .catch((reason) => reject(reason));
-        }).then((fee) => {
+        });
+    };
+
+    beforeEach(async () => {
+        process.env.NODE_ENV = 'test';
+
+        const provider = new WsProvider('ws://127.0.0.1:9944');
+        api = await ApiPromise.create({ provider });
+        const keyring = new Keyring({ type: 'sr25519' });
+
+        lender = keyring.addFromUri('//Alice', { name: 'Alice' });
+        const lenderAddress = randomEthAddress();
+
+        borrower = keyring.addFromUri('//Bob', { name: 'Bob' });
+        const borrowerAddress = randomEthAddress();
+        [lenderRegAddr, borrowerRegAddr] = await Promise.all([
+            testUtils.registerAddress(api, lenderAddress, blockchain, lender),
+            testUtils.registerAddress(api, borrowerAddress, blockchain, borrower),
+        ]);
+    }, 60000);
+
+    afterEach(async () => {
+        await api.disconnect();
+    });
+
+    it('fee is min 0.01 CTC', async (): Promise<void> => {
+        return getRegisterDealOrderFee().then((fee) => {
             expect(fee).toBeGreaterThanOrEqual(POINT_01_CTC);
         });
     }, 30000);
+
+    it('fee scales linearly', async (): Promise<void> => {
+        // WARNING: don't use Promise.all() here b/c it doesn't guarantee
+        // execution order
+        const fee01 = await getRegisterDealOrderFee();
+console.log("***** after fee01", fee01);
+
+        const fee02 = await getRegisterDealOrderFee();
+console.log("***** after fee02", fee02);
+
+        const fee03 = await getRegisterDealOrderFee();
+console.log("***** after fee03", fee03);
+
+        const fee04 = await getRegisterDealOrderFee();
+console.log("***** after fee04", fee04);
+
+        const fee05 = await getRegisterDealOrderFee();
+console.log("***** after fee05", fee05);
+
+        const fee06 = await getRegisterDealOrderFee();
+console.log("***** after fee06", fee06);
+
+        const fee07 = await getRegisterDealOrderFee();
+console.log("***** after fee07", fee07);
+
+        const fee08 = await getRegisterDealOrderFee();
+console.log("***** after fee08", fee08);
+
+        const fee09 = await getRegisterDealOrderFee();
+console.log("***** after fee09", fee09);
+
+        const fee10 = await getRegisterDealOrderFee();
+console.log("***** after fee10", fee10);
+
+        expect(fee01).toBeLessThan(fee10);
```

Comment:
@nathanwhit this is failing b/c `fee10` < `fee01`. The results I get locally are:
```
    ***** after fee01 22998375104903137n
    ***** after fee02 22998105151999554n
    ***** after fee03 22997835203144924n
    ***** after fee04 22997565258339186n
    ***** after fee05 22997295317582279n
    ***** after fee06 22997025380874142n
    ***** after fee07 22996755448214716n
    ***** after fee08 22996485519603938n
    ***** after fee09 22996215595041749n
    ***** after fee10 22995945674528087n
```

You can see that after every subsequent call to RegisterDealOrder the fee actually decreases slightly.  Doesn't sound right to me.

Reply:
Sorry, I haven't been quite precise about this. How it works is that we target a certain block fullness (25%), so we adjust fees downward if blocks are less than 25% full, and adjust fees upwards if blocks are >25% full. It's derived from [this research](https://research.web3.foundation/en/latest/polkadot/overview/2-token-economics.html#-2.-slow-adjusting-mechanism) and it matches what polkadot (and most other substrate chains I could find) uses. At the current settings fees can change only up to ~3% across a day's worth of blocks.